### PR TITLE
Save interpretationRules on TestType creation and expose update TestType API

### PR DESCRIPTION
**Pull request by kstam**
**When applied this will**
* Properly read interpretation rules during test type creation and persist them to the database
* Start sending a simplified testType version nested in each test (that contains only name, id and permissions). This way we don't have to send the quite bulky interpretation rules and results schema back with every test.
* Expose a `PATCH /test-types/:id` API to allow updating existing test-types. 
* Fix results not being updated when they already exist
*  Store AdministeringUserId with test so that we know which user administered the test

### CODE REVIEW

File: src/api/transformers/test/transfromTestToDTO.ts [hunk 1]
```diff
@@ -3,15 +3,19 @@ import { TestDTO } from '../../interface';
...
 import { transformTestInterpretationToDTO } from './transformTestInterpretationToDTO';
+import { transformTestTypeToSimpleDTO } from './transformTestTypeToSimpleDTO';
 
 export function transformTestToDTO(test: Test): TestDTO {
   return {
     id: test.id.value,
     userId: test.userId.value,
     creationTime: test.creationTime,
+    administeredBy: {
+      userId: test.administeredBy.value,
```
Comment (StevePole) [inline src/api/transformers/test/transfromTestToDTO.ts:14]: Why did you go nested for this one?  If we're going to nest, it makes me wonder if the administration confidence should be nested together as an administration section?

File: src/application/service/tests/UpdateTestType.test.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  it('throws not found error if test type id does not exist', async () => {
+    const testTypeId = new TestTypeId().value;
+    await expect(updateTestType.execute(testTypeId, { name: 'NEW_NAME' })).rejects.toEqual(
+      new TestTypeNotFoundError(testTypeId)
+    );
+  });
+
+  it('updates the test type name', async () => {
+    const testType = await testTypeRepository.save(aTestType());
+    const newName = v4();
```
Comment (StevePole) [inline src/application/service/tests/UpdateTestType.test.ts:27]: Little bit weird?  Introduces non-determinism into the test, I know it doesn't (shouldn't...) matter in this case, but maybe just simpler all round to hard code a string?

File: src/api/transformers/test/transfromTestToDTO.ts (hunk 1 above)
Comment (kstam) [reply src/api/transformers/test/transfromTestToDTO.ts:14]: I went for nested as we might need to add name information there (Results Entered By, Administered by) We can move the confidence in there too.
Something like:
```
administration: {
   userId: ``,
   confidence: ``,
}
```

And same for results:

```
entry: {
    userId: ``
    confidence: ``
}
```

File: src/application/service/tests/UpdateTestType.test.ts (hunk 2 above)
Comment (kstam) [reply src/application/service/tests/UpdateTestType.test.ts:27]: Ha! I knew someone would catch this :) Question: isn't there less non-determinism? As in, isn't this a bigger guarantee that the name will be a unique string vs a hard coded value (which might colide with whatever the `aTestType()` factory returns?

File: src/database/migrations/V1__addTest.ts [hunk 3]
```diff
@@ -7,6 +7,7 @@ export async function up(db: knex) {
   await db.schema.createTable(TEST_TABLE, (table) => {
     table.uuid('id').primary();
     table.uuid('user_id').notNullable();
+    table.uuid('administering_user_id').notNullable();
```
Comment (StevePole) [inline src/database/migrations/V1__addTest.ts:10]: In the test results table, we have a column performing a similar task, but it's called `created_id`, just wondering if they should share a more similar name?

Comment (kstam) [reply src/database/migrations/V1__addTest.ts:10]: Good point. Maybe we should unify all of this as:

```
creator_user_id: ``
creator_confidence: ``
```

Then on the API layer we can rename `creation` to `administration` for `test` and `entry` for `result`

Comment (StevePole) [reply src/database/migrations/V1__addTest.ts:10]: What about if they're both:
```
creator: {
  user_id: ...,
  confidence: ...
}
```
Feels a bit weird to have the same content under different keys?

Comment (kstam) [reply src/database/migrations/V1__addTest.ts:10]: Yeah.. this is true... let's go with this as it is simpler.. I'll change accrodingly

**Pull request merged**